### DISCUSSION TITLE
Error: Resource Not Found for /element/{elementId} Endpoint When Accessing Element Attributes

### DISCUSSION THREAD

**Discussion by sidjdev** [appium/appium#19731] [Q&A] [answered]
While running my iOS automation tests using Appium, I am trying to retrieve attributes of a UI element (like UID, accessibilityContainer, accessible, focused, frame, hittable, index, label, type, value, visible). However, when I send a request to the endpoint /wd/hub/session/{sessionId}/element/{elementId}/, I receive the following error:

`The requested resource could not be found, or a request was received using an HTTP method that is not supported by the mapped resource`

Appium version: 2.4.1
Desktop OS/version used to run Appium: macOS Sonoma
Mobile platform/version under test: iOS 17.2
Real device or emulator/simulator: Simulator

**Comment by jlipps** [comment]
What commands did you use to start the session and find the element before this?

**Reply by sidjdev** [reply]
Hi. I used `"http://192.168.0.102:4723/session"` to start the session and `"\(sessionId)/element"` to get the element

**Reply by KazuCocoa** [reply]
Could you share the full appium log (as GitHub GIST)?

**Reply by sidjdev** [reply]
[Terminal response](https://gist.github.com/sidjdev/666314ecc8a8c037abef79d158f62109)

Is this what you mean? This is my repose in Terminal

[Xcode Log](https://gist.github.com/sidjdev/c75518dc96902121de48e2de16d5e188)

This is the response I am getting(Xcode log)

**Reply by KazuCocoa** [reply]
No, I mean appium server logs started from:
```
[Appium] Welcome to Appium v2.4.1 (REV 552db40622bb7a82d9c6d67d2d6bcf3694b47e30)
[Appium] The autodetected Appium home path: /Users/kazu/.appium
[Appium] Attempting to load driver xcuitest...
[Appium] Requiring driver at /Users/kazu/.appium/node_modules/appium-xcuitest-driver/build/index.js
[Appium] XCUITestDriver has been successfully loaded in 4.662s
[Appium] Appium REST http interface listener started on http://0.0.0.0:4723
[Appium] You can provide the following URLs in your client code to connect to this server:
[Appium] 	http://127.0.0.1:4723/ (only accessible from the same host)
[Appium] 	http://192.168.4.248:4723/
[Appium] 	http://172.16.25.21:4723/
[Appium] Available drivers:
[Appium]   - xcuitest@5.14.1 (automationName 'XCUITest')
[Appium] No plugins have been installed. Use the "appium plugin" command to install the one(s) you want to use.
```
(appium 2 and xcuitest driver case)

**Reply by KazuCocoa** [reply, marked as answer]
btw, https://gist.github.com/sidjdev/666314ecc8a8c037abef79d158f62109 are expected since no such routes exist. They should be  `/session/:sessionId/element/:elementId/property/:name` or `/session/:sessionId/element/:elementId/attribute/:name` usually. Clients may handle them though.

**Reply by sidjdev** [reply]
Than you for your help. While I have you here, could you also tell me if it's possible to run Appium against an app that is installed through TestFlight?

**Reply by KazuCocoa** [reply]
I haven't tried via TestFlight, but perhaps it is not possible because basically the XCTest package probably cannot create an archive to distribute via TestFlight.

**Reply by jlipps** [reply]
if the app is already on the phone, shouldn't it be possible to just start a WDA session, navigate to the app and then have automation capabilities that way?

**Comment by PriscillaMul** [comment]
I am try to automate the appium sever instead of doing it on command line manually but I am getting below infinite error

[Appium] Welcome to Appium v2.15.0
[Appium] Non-default server args:
[Appium] { address: '127.0.0.1', sessionOverride: true }
[Appium] The autodetected Appium home path: C:\Users\\.appium
[Appium] Attempting to load driver uiautomator2...
[Appium] Requiring driver at C:\Users\\.appium\node_modules\appium-uiautomator2-driver\build\index.js
[Appium] AndroidUiautomator2Driver has been successfully loaded in 2.055s
[Appium] Appium REST http interface listener started on http://127.0.0.1:4723
[Appium] Available drivers:
[Appium]   - uiautomator2@4.0.1 (automationName 'UiAutomator2')
[Appium] Available plugins:
[Appium]   - execute-driver@4.0.1
[Appium]   - images@3.0.27
[Appium]   - relaxed-caps@1.0.6
[Appium]   - universal-xml@1.0.27
[Appium] No plugins activated. Use the --use-plugins flag with names of plugins to activate
[HTTP] --> GET /wd/hub/status {}
[HTTP] No route found for /wd/hub/status
[HTTP] <-- GET /wd/hub/status 404 16 ms - 211 
[HTTP] --> GET /wd/hub/status {}
[HTTP] No route found for /wd/hub/status
[HTTP] <-- GET /wd/hub/status 404 8 ms - 211 
[HTTP] --> GET /wd/hub/status {}
[HTTP] No route found for /wd/hub/status
[HTTP] <-- GET /wd/hub/status 404 1 ms - 211 
[HTTP] --> GET /wd/hub/status {}
[HTTP] No route found for /wd/hub/status
[HTTP] <-- GET /wd/hub/status 404 1 ms - 211 
[HTTP] --> GET /wd/hub/status {}
[HTTP] No route found for /wd/hub/status
[HTTP] <-- GET /wd/hub/status 404 2 ms - 211 
[HTTP] --> GET /wd/hub/status {}
[HTTP] No route found for /wd/hub/status
[HTTP] <-- GET /wd/hub/status 404 2 ms - 211 
[HTTP] --> GET /wd/hub/status {}
[HTTP] No route found for /wd/hub/status
[HTTP] <-- GET /wd/hub/status 404 1 ms - 211 
[HTTP] --> GET /wd/hub/status {}
[HTTP] No route found for /wd/hub/status
[HTTP] <-- GET /wd/hub/status 404 2 ms - 211

**Reply by KazuCocoa** [reply]
I guess your client kept asking `/wd/hub`. Then, `--base-path=/wd/hub` for the appium server flag would help
